app/src/main/java/org/sopt/sample/presentation/signin/SignInActivity.kt

```diff
@@ -0,0 +1,80 @@
+package org.sopt.sample.presentation.signin
+
+import android.content.Intent
+import android.os.Build
+import androidx.appcompat.app.AppCompatActivity
+import android.os.Bundle
+import androidx.activity.result.ActivityResultLauncher
+import androidx.activity.result.contract.ActivityResultContracts
+import androidx.annotation.RequiresApi
+import org.sopt.sample.R
+import org.sopt.sample.data.User
+import org.sopt.sample.databinding.ActivitySignInBinding
+import org.sopt.sample.presentation.main.MyPageActivity
+import org.sopt.sample.presentation.signup.SignUpActivity
+import org.sopt.sample.util.showSnackBar
+import org.sopt.sample.util.showToast
+
+
+
+class SignInActivity : AppCompatActivity() {
+    private lateinit var binding: ActivitySignInBinding
+    private var user: User? = null
+    private lateinit var resultLauncher: ActivityResultLauncher<Intent>
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        binding = ActivitySignInBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+        onSignUpClick()
+        onSignInClick()
+        setSignUp()
+    }
+
+    private fun onSignUpClick() {
+        binding.btnSignUp.setOnClickListener {
+            val signUpIntent = Intent(this, SignUpActivity::class.java)
+            resultLauncher.launch(signUpIntent)
+        }
+    }
+
+    private fun onSignInClick() {
+        binding.btnSignIn.setOnClickListener {
+            if (isSuccessLogin()) {
+                goToMyPage()
+            } else {
+                showToast("로그인 실패했습니다.")
+            }
+        }
+    }
+
+    private fun setSignUp() {
+        resultLauncher =
+            registerForActivityResult(ActivityResultContracts.StartActivityForResult()) { result ->
+                if (result.resultCode == RESULT_OK) {
+                    if (result.data?.hasExtra("user") == true) {
+                        showSnackBar(binding.root, getString(R.string.signup_complete))
+                        user = result.data?.getParcelableExtra(User)
+                    }
+                    binding.apply {
+                        etId.setText(user?.id)
+                        etPw.setText(user?.password)
+                    }
```

Comment:
```suggestion
                    with(binding){
                        etId.setText(user?.id)
                        etPw.setText(user?.password)
                    }
```
이렇게는 어떠신가요?! apply로 반환받은 수신 객체를 사용할 것이 아니라면 with 사용이 더 좋아보입니다 :) 그리고 내부적으로 editText의 text 속성을 변경하지만,,! 겉으로 보기에는 setText() 함수에 인자를 넘겨주는 거라 프로퍼티의 속성을 바꿔주는 것처럼 보여지지 않아서 어색하게 느껴지는 것 같습니다 🧐 사실 이 코멘트는 패스하셔도 상관없긴 합니다~!

Reply:
아주~! 좋습니다~! 사실 팀원들에게 with() 쓰라고 했으면서 ㅋㅋㅋ 제가 apply를 썼네요,, 로직 변경하면서 반영하겠습니다 :)